### PR TITLE
Bug fixed for MyAvatar.getJointRotation(name, quat) and MyAvatar.setJointRotation(name, quat)

### PR DESCRIPTION
**Pull request by amantley**
This pr addresses the bug where the getJointRotation and setJointRotation are not the same when you use different overloads of the function.  Specifically, the MyAvatar.getJointRotation(index,quat) is working but  MyAvatar.getJointRotation("jointname", quat) is not. 

 https://highfidelity.manuscript.com/f/cases/21542/MyAvatar-getJointTranslation-index-and-getJointTranslation-name-return-different-values

### Test Plan
1)  open Interface and run the following script.
https://hifi-content.s3.amazonaws.com/angus/getJointTranslationRotation.js
2) expected result: the second and third outputs for the translations and rotations should match

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18558/HighFidelity-Beta-PR15316-857a4ab.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18558/HighFidelity-Quest-Beta-PR15316-857a4ab.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18558/HighFidelity-Beta-PR15316-857a4ab.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18558/HighFidelity-Beta-Interface-PR15316-857a4ab.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18558/HighFidelity-Beta-PR15316-857a4ab.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18558/HighFidelity-Beta-Interface-PR15316-857a4ab.exe

File: libraries/avatars/src/AvatarData.h [hunk 1]
```diff
@@ -1750,14 +1750,14 @@ public slots:
...
     template <typename T, typename F>
     T readLockWithNamedJointIndex(const QString& name, const T& defaultValue, F f) const {
-        int index = getFauxJointIndex(name);
-        QReadLocker readLock(&_jointDataLock);
-
-        // The first conditional is superfluous, but illustrative
-        if (index == -1 || index < _jointData.size()) {
+        int index = getJointIndex(name);
+        if (index == -1) {
+            index = getFauxJointIndex(name);
```
Comment (SimonWalton-HiFi) [inline libraries/avatars/src/AvatarData.h]: It looks like `Avatar::getJointIndex()` handles the faux joints so this shouldn't be required.

File: libraries/avatars/src/AvatarData.cpp [hunk 2]
```diff
@@ -1721,7 +1721,6 @@ glm::vec3 AvatarData::getJointTranslation(const QString& name) const {
     // on another thread in between the call to getJointIndex and getJointTranslation
     // return getJointTranslation(getJointIndex(name));
```
Comment (SimonWalton-HiFi) [inline libraries/avatars/src/AvatarData.cpp]: The (existing) comments talks about locking covering getJointIndex & getJointTranslation but the implementation of `readLockWithNamedJointIndex` doesn't actually lock 'til after looking up the name. Should the lock be moved?

Comment (amantley) [reply libraries/avatars/src/AvatarData.cpp]: Yes I'll fix that

File: libraries/avatars/src/AvatarData.h (hunk 1 above)
Comment (amantley) [reply libraries/avatars/src/AvatarData.h]: I think Avatar::getJointIndex(name) handles all joints
int Avatar::getJointIndex(const QString& name) const {
    int result = getFauxJointIndex(name);
    **if (result != -1) {**
        return result;
    }

    withValidJointIndicesCache([&]() {
        if (_modelJointIndicesCache.contains(name)) {
            result = _modelJointIndicesCache[name] - 1;
        }
    });
    return result;
}

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18577/HighFidelity-Beta-PR15316-9afbf76.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18577/HighFidelity-Beta-Interface-PR15316-9afbf76.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18577/HighFidelity-Beta-PR15316-9afbf76.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18577/HighFidelity-Quest-Beta-PR15316-9afbf76.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18577/HighFidelity-Beta-PR15316-9afbf76.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18577/HighFidelity-Beta-Interface-PR15316-9afbf76.exe

**Pull request merged**